be/src/pipeline/pipeline_x/dependency.h

```diff
@@ -575,7 +577,9 @@ struct HashJoinSharedState : public JoinSharedState {
     bool probe_ignore_null = false;
 };
 
-struct PartitionedHashJoinSharedState : public HashJoinSharedState {
+struct PartitionedHashJoinSharedState
+        : public HashJoinSharedState,
+          public std::enable_shared_from_this<PartitionedHashJoinSharedState> {
     std::vector<std::unique_ptr<vectorized::MutableBlock>> partitioned_build_blocks;
```

Comment:
please add enable_factory_creator for this object to forbidden raw pointer for this object。
And also using class not struct. I am not sure if enable_factory_creator has effect on struct.